styx-common/src/main/java/com/spotify/styx/serialization/PersistentEvent.java

```diff
@@ -89,8 +90,8 @@ public PersistentEvent created(WorkflowInstance workflowInstance, String executi
     }
 
     @Override
-    public PersistentEvent dequeue(WorkflowInstance workflowInstance) {
-      return new PersistentEvent("dequeue", workflowInstance.toKey());
+    public PersistentEvent dequeue(WorkflowInstance workflowInstance, Set<String> resourceIds) {
```

Comment:
We might need to make `resourceIds` as `Optional<Set<String>>`. Backfill still replays events.

Reply:
We did exactly the same thing before, so something we can refer to https://github.com/spotify/styx/commit/6c3ce31a269f6e0f555875776a05eebebeee3ada#diff-caa8e641522b9a838652711092851507R92

Reply:
In a bunch of places `null` was intentionally used to test backwards compatibility. (Deserialization when resource_ids is absent.)  I think the latest changes have made it actually different from the handling in other cases... Confused.